src/cdk-experimental/listbox/listbox.ts

```diff
@@ -584,6 +655,50 @@ export class CdkListbox<T = unknown>
     }
   }
 
+  /**
+   * Trigger the given range of options in response to user interaction.
+   * Should only be called in multi-selection mode.
+   * @param trigger The option that was triggered
+   * @param from The start index of the options to toggle
+   * @param to The end index of the options to toggle
+   * @param on Whether to toggle the option range on
+   */
+  protected triggerRange(trigger: CdkOption<T> | null, from: number, to: number, on: boolean) {
+    if (this.disabled || (trigger && trigger.disabled)) {
+      return;
+    }
+    this._lastTriggered = trigger;
+    const isEqual = this.compareWith ?? Object.is;
+    const options = this.options.toArray();
+    const updateValues = options
+      .slice(Math.max(0, Math.min(from, to)), Math.min(options.length, Math.max(from, to) + 1))
+      .map(option => option.value);
+    const selected = [...this.value];
+    for (const updateValue of updateValues) {
+      const selectedIndex = selected.findIndex(selectedValue =>
+        isEqual(selectedValue, updateValue),
+      );
+      if (on && selectedIndex === -1) {
+        selected.push(updateValue);
+      } else if (!on && selectedIndex !== -1) {
+        selected.splice(selectedIndex, 1);
+      }
+    }
+    let changed = false;
+    this.selectionModel.changed
+      .pipe(take(1), takeUntil(this.destroyed))
+      .subscribe(() => (changed = true));
```

Comment:
Yeah this definitely feels weird, now that I've extended `SelectionModel` maybe I can simplify it. Regarding the potential leak, I originally had this as a `.unsubscribe()` that happened regardless of if it emitted. `take(1)` was your suggestion 😛

Reply:
Hmm actually, I don't think its easy to change in just the `ListboxSelectionModel` I'd have to change a few `void` return functions to `boolean` on `SelectionModel`. Does that count as a breaking change? It feels like it can probably just be a minor change...

Side note: I don't really like `SelectionModel`, I feel like I've been fighting against it the whole time I've been trying to implement this listbox...

Reply:
I meant that we can do it in the listbox-specific selection model which is internal.

Reply:
That's not really possible the way the code is currently structured, because the call to the super function emits the event and clears out the pending updates. I guess I could restructure the code to allow me to make the API change only in the ListboxSelectionModel, but it seems like a useful change for the SelectionModel and should be non-breaking